### ISSUE TITLE
Can't filter by third party name if they contains ()

### ISSUE THREAD

**Issue by Elikill58** [Dolibarr/dolibarr#28188]
### Bug

By calling API with URL like that:
```php
$sqlFilter = urlencode("t.nom:=:'myThirdarty (MT)'");
callApi("api/index.php/thirdparties?sqlfilters=$sqlFilter");
```
You will get error `Bad Request: Error when validating parameter sqlfilters -> Bad syntax of the search string`.

### Environment Version

18.0.5

### Environment OS

Debian 12

### Environment Web server

Apache 2

### Environment PHP

8.1

### Environment Database

mysql

### Expected and actual behavior

I were using version 15.0.3. I'm actualling migrating. I want to get a third party by the login/name.

### Steps to reproduce the behavior

1) Create user with parenthesis in his name like `My name (MM)`
2) Call API with URL like that: `api/index.php/thirdparties?sortfield=t.rowid&sortorder=ASC&limit=100&sqlfilters=t.nom%3A%3D%3A'MyName%20(MN)'`

### Attached files

Example of full query using swagger (I removed unused line) :
![image](https://github.com/Dolibarr/dolibarr/assets/17381066/95358886-796b-4ed6-a606-e39959ce4ffc)

**Comment by JonBendtsen** [comment]
Interesting issue. Have you tried using the CHAR(40) solution mentioned on this page?

https://salesforce.stackexchange.com/questions/383197/escaping-parenthesis-in-sql-query-like-clause

**Comment by Elikill58** [comment]
I just tried it and it the same result:
![image](https://github.com/Dolibarr/dolibarr/assets/17381066/d4c58dc9-726c-4b60-998d-fe7ddc53b7d6)

**Comment by d3v2a** [comment]
This will work but return some unwanted result  (`_` is any character in MySQL)

```php
$sqlFilter = urlencode("t.nom:=:'myThirdarty _MT_'");
```

**Comment by hregis** [comment]
and with just : `(t.nom:=:'myThirdarty \(MT\)')`

**Comment by Elikill58** [comment]
> This will work but return some unwanted result (`_` is any character in MySQL)
> 
> ```
> $sqlFilter = urlencode("t.nom:=:'myThirdarty _MT_'");
> ```

We should use `LIKE` with this method (instead of =) ?

**Comment by Elikill58** [comment]
> and with just : `(t.nom:=:'myThirdarty \(MT\)')`

No, it doesn't work. I tried lot of example as showed in the link of StackExchange from Jon and it's blcoked too

**Comment by d3v2a** [comment]
> We should use `LIKE` with this method (instead of =) ?

It works with `=` and is more close to the expected result (`_` is only one of any character, `%` is one or many characters)

**Comment by Elikill58** [comment]
> > We should use `LIKE` with this method (instead of =) ?
> 
> It works with `=` and is more close to the expected result (`_` is only one of any character, `%` is one or many characters)

I just tried and for me it doesn't works with =.

- With `t.nom:=:'myThirdarty _MT_'` -> no result found. So it's better, I no longer have error.
- With `t.nom:LIKE:'myThirdarty _MT_'` -> Result found
- I try by adding `myThirdarty MT` and `myThirdarty aMTa` to see the result, and try `t.nom:LIKE:'myThirdarty _MT_'` and I got 2 result : The right one (with `(MT)`) and with a (`aMTa`)

**Comment by caos30** [comment]
I’m facing a similar problem. In my case, I need to retrieve third-party records with phone numbers formatted like `(1) 234-56-78` or similar variations (e.g., with parentheses, dashes, or spaces).

After some testing, the following expression worked for me:
```
(t.phone:like:'%1%2%3%4%5%6%7%8%')  
```

**Advantage:** This approach matches the phone number regardless of formatting differences (parentheses, dashes, or spaces), as it searches for the consecutive chain of digits in the field.

For example, the above would match any record containing 12345678 in the phone field, even if stored as (123) 456-78, 123-45-678, etc.

**Note:** i need this search to be used by n8n to retrieve existing third parties given a phone number incoming from a marketing funnel form. If it doesn't exist then we will create a new third party.

Hope this helps! 😊
